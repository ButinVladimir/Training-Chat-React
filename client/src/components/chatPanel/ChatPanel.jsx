import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import ErrorPanel from './errorPanel/ErrorPanelContainer';
import MessagesList from './messagesList/MessagesListContainer';

function ChatPanel({ showError, socket }) {
  return (
    <Fragment>
      {showError && <ErrorPanel />}
      {!showError && <MessagesList socket={socket} />}
    </Fragment>
  );
}

ChatPanel.propTypes = {
  showError: PropTypes.bool.isRequired,
  socket: PropTypes.any.isRequired,
};

export default ChatPanel;
