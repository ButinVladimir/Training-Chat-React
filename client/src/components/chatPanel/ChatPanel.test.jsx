import React from 'react';
import { Socket } from 'socket.io-client';
import { shallow } from 'enzyme';
import ChatPanel from './ChatPanel';

describe('ChatPanel', () => {
  let mockSocket;

  beforeEach(() => {
    mockSocket = new Socket();
  });

  it('renders without crashing when error is shown', () => {
    const component = shallow(<ChatPanel showError socket={mockSocket} />);
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when connected', () => {
    const component = shallow(
      <ChatPanel showError={false} socket={mockSocket} />,
    );
    expect(component).toMatchSnapshot();
  });
});
