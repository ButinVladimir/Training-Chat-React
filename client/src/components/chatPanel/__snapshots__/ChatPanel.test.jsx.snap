// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPanel renders without crashing when connected 1`] = `
<Fragment>
  <Connect(MessagesList)
    socket={lookup {}}
  />
</Fragment>
`;

exports[`ChatPanel renders without crashing when error is shown 1`] = `
<Fragment>
  <Connect(ErrorPanel) />
</Fragment>
`;
